src/ir/named.rs

```diff
@@ -359,33 +359,47 @@ impl<'ctx, 'gen> MonotoneFramework for UsedTemplateParameters<'ctx, 'gen>  {
     }
 
     fn constrain(&mut self, id: ItemId) -> bool {
-        let original_len = self.used[&id].len();
+        // Invariant: all hash map entries' values are `Some` upon entering and
+        // exiting this method.
+        debug_assert!(self.used.values().all(|v| v.is_some()));
```

Comment:
> It will only run in debug builds, which I don't think anyone is using outside hacking on bindgen itself, so I think we're fine.

You forget everyone doing a debug build with build-time bindgen, unfortunately :(